[CI] ShardBulkInferenceActionFilterIT testItemFailures {p0=true p1=true} failing

**Build Scans:**
- [elasticsearch-intake #33336 / part2](https://gradle-enterprise.elastic.co/s/zt6eqk72zrck2)
- [elasticsearch-pull-request #114618 / part-2](https://gradle-enterprise.elastic.co/s/xf52aq6n7by46)
- [elasticsearch-pull-request #114396 / part-2](https://gradle-enterprise.elastic.co/s/x2xogxaapsqry)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:internalClusterTest" --tests "org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.testItemFailures {p0=true p1=true}" -Dtests.seed=D69C2DB07CC01D10 -Dtests.locale=cgg -Dtests.timezone=Africa/Gaborone -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testItemFailures%20%7Bp0%3Dtrue%20p1%3Dtrue%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "expected [String|Number|Boolean]"
     but: was "org.elasticsearch.transport.RemoteTransportException: [node_s0][127.0.0.1:23023][indices:data/write/bulk[s]]"
```

**Issue Reasons:**
- [main] 3 failures in test testItemFailures {p0=true p1=true} (1.1% fail rate in 274 executions)
- [main] 2 failures in step part-2 (1.0% fail rate in 191 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.1% fail rate in 184 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -345,30 +345,9 @@ tests:
 - class: org.elasticsearch.compute.aggregation.AllLastBytesRefByTimestampGroupingAggregatorFunctionTests
   method: testSimpleWithCranky
   issue: https://github.com/elastic/elasticsearch/issues/140763
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testItemFailures {p0=true p1=true}
-  issue: https://github.com/elastic/elasticsearch/issues/140766
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testBulkOperations {p0=true p1=true}
-  issue: https://github.com/elastic/elasticsearch/issues/140767
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testBulkOperations {p0=false p1=true}
-  issue: https://github.com/elastic/elasticsearch/issues/140768
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testItemFailures {p0=false p1=true}
-  issue: https://github.com/elastic/elasticsearch/issues/140769
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testBulkOperations {p0=true p1=false}
-  issue: https://github.com/elastic/elasticsearch/issues/140772
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testItemFailures {p0=true p1=false}
-  issue: https://github.com/elastic/elasticsearch/issues/140773
 - class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
   method: test {p0=esql/46_downsample/Stats from downsampled and non-downsampled index simultaneously with implicit casting}
   issue: https://github.com/elastic/elasticsearch/issues/140774
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testItemFailures {p0=false p1=false}
-  issue: https://github.com/elastic/elasticsearch/issues/140778
 - class: org.elasticsearch.compute.aggregation.AllFirstBytesRefByTimestampGroupingAggregatorFunctionTests
   method: testSimpleWithCranky
   issue: https://github.com/elastic/elasticsearch/issues/140784

--- a/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java
+++ b/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java
@@ -78,7 +78,6 @@ import java.util.Iterator;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.stream.Collectors;
 
 import static java.util.Collections.singletonList;
@@ -259,7 +258,6 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
         private final Runnable onCompletion;
         private final AtomicArray<FieldInferenceResponseAccumulator> inferenceResults;
         private final IndexingPressure.Coordinating coordinatingIndexingPressure;
-        private final Map<FailureSignature, Exception> deduplicatedFailures;
 
         private AsyncBulkShardInferenceAction(
             boolean useLegacyFormat,
@@ -274,7 +272,6 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
             this.inferenceResults = new AtomicArray<>(bulkShardRequest.items().length);
             this.onCompletion = onCompletion;
             this.coordinatingIndexingPressure = coordinatingIndexingPressure;
-            this.deduplicatedFailures = new HashMap<>();
         }
 
         @Override
@@ -565,7 +562,7 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
                          * This ensures that the field is treated as intentionally cleared,
                          * preventing any unintended carryover of prior inference results.
                          */
-                        if (incrementIndexingPressurePreInference(indexRequest, itemIndex) == false) {
+                        if (incrementIndexingPressure(indexRequest, itemIndex) == false) {
                             return inputLength;
                         }
 
@@ -603,7 +600,7 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
                     List<FieldInferenceRequest> requests = requestsMap.computeIfAbsent(inferenceId, k -> new ArrayList<>());
                     int offsetAdjustment = 0;
                     for (String v : values) {
-                        if (incrementIndexingPressurePreInference(indexRequest, itemIndex) == false) {
+                        if (incrementIndexingPressure(indexRequest, itemIndex) == false) {
                             return inputLength;
                         }
 
@@ -651,7 +648,7 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
             }
         }
 
-        private boolean incrementIndexingPressurePreInference(IndexRequestWithIndexingPressure indexRequest, int itemIndex) {
+        private boolean incrementIndexingPressure(IndexRequestWithIndexingPressure indexRequest, int itemIndex) {
             boolean success = true;
             if (indexRequest.isIndexingPressureIncremented() == false) {
                 try {
@@ -696,7 +693,9 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
          */
         private void applyInferenceResponses(BulkItemRequest item, FieldInferenceResponseAccumulator response) throws IOException {
             if (response.failures().isEmpty() == false) {
-                handleInferenceFailures(item, response.failures());
+                for (var failure : response.failures()) {
+                    item.abort(item.index(), failure);
+                }
                 return;
             }
 
@@ -749,21 +748,6 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
                 inferenceFieldsMap.put(fieldName, result);
             }
 
-            updateIndexSource(item, inferenceFieldsMap);
-        }
-
-        private void handleInferenceFailures(BulkItemRequest item, List<Exception> failures) {
-            for (Exception failure : failures) {
-                // Generate a signature for the failure to deduplicate on the most important properties
-                FailureSignature failureSignature = new FailureSignature(failure);
-
-                Exception deduplicatedFailure = deduplicatedFailures.computeIfAbsent(failureSignature, k -> failure);
-                item.abort(item.index(), deduplicatedFailure);
-            }
-        }
-
-        private void updateIndexSource(BulkItemRequest item, Map<String, Object> inferenceFieldsMap) throws IOException {
-            IndexRequest indexRequest = getIndexRequestOrNull(item.request());
             IndexSource indexSource = indexRequest.indexSource();
             int originalSourceSize = indexSource.byteLength();
             BytesReference originalSource = indexSource.bytes();
@@ -845,34 +829,6 @@ public class ShardBulkInferenceActionFilter implements MappedActionFilter {
         @Override
         public Iterator<Chunk> chunksAsByteReference(XContent xcontent) {
             return Collections.emptyIterator();
-        }
-    }
-
-    static class FailureSignature {
-        private final Class<? extends Throwable> failureClass;
-        private final String failureMessage;
-        private final FailureSignature failureCauseSignature;
-
-        FailureSignature(Throwable failure) {
-            failureClass = failure.getClass();
-            failureMessage = failure.getMessage();
-            failureCauseSignature = failure.getCause() != null ? new FailureSignature(failure.getCause()) : null;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            FailureSignature that = (FailureSignature) o;
-
-            return Objects.equals(failureClass, that.failureClass)
-                && Objects.equals(failureMessage, that.failureMessage)
-                && Objects.equals(failureCauseSignature, that.failureCauseSignature);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(failureClass, failureMessage, failureCauseSignature);
         }
     }
 }

--- a/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilterTests.java
+++ b/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilterTests.java
@@ -87,9 +87,6 @@ import java.util.Set;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.function.BiFunction;
-import java.util.function.Consumer;
 
 import static org.elasticsearch.common.bytes.BytesReferenceTestUtils.equalBytes;
 import static org.elasticsearch.index.IndexingPressure.MAX_COORDINATING_BYTES;
@@ -110,9 +107,7 @@ import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.notNullValue;
-import static org.hamcrest.Matchers.sameInstance;
 import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.ArgumentMatchers.anyString;
@@ -732,8 +727,7 @@ public class ShardBulkInferenceActionFilterTests extends ESTestCase {
     public void testIndexingPressureTripsOnInferenceRequestGeneration() throws Exception {
         final InferenceStats inferenceStats = InferenceStatsTests.mockInferenceStats();
         final InstrumentedIndexingPressure indexingPressure = new InstrumentedIndexingPressure(
-            // Set the max coordinating bytes to a value smaller than the source size
-            Settings.builder().put(MAX_COORDINATING_BYTES.getKey(), "15kb").build()
+            Settings.builder().put(MAX_COORDINATING_BYTES.getKey(), "1b").build()
         );
         final StaticModel sparseModel = StaticModel.createRandomInstance(TaskType.SPARSE_EMBEDDING);
         final ShardBulkInferenceActionFilter filter = createFilter(
@@ -744,9 +738,7 @@ public class ShardBulkInferenceActionFilterTests extends ESTestCase {
             inferenceStats
         );
 
-        // Generate a very large field value to trigger the indexing pressure failure
-        String sparseFieldValue = randomAlphanumericOfLength(50000);
-        XContentBuilder doc1Source = IndexSource.getXContentBuilder(XContentType.JSON, "sparse_field", sparseFieldValue);
+        XContentBuilder doc1Source = IndexSource.getXContentBuilder(XContentType.JSON, "sparse_field", "bar");
 
         CountDownLatch chainExecuted = new CountDownLatch(1);
         ActionFilterChain<BulkShardRequest, BulkShardResponse> actionFilterChain = (task, action, request, listener) -> {
@@ -1011,152 +1003,6 @@ public class ShardBulkInferenceActionFilterTests extends ESTestCase {
         IndexingPressure.Coordinating coordinatingIndexingPressure = indexingPressure.getCoordinating();
         assertThat(coordinatingIndexingPressure, notNullValue());
         verify(coordinatingIndexingPressure).close();
-    }
-
-    @SuppressWarnings("unchecked")
-    public void testDeduplicatedFailures() throws Exception {
-        final InferenceStats inferenceStats = InferenceStatsTests.mockInferenceStats();
-        final InstrumentedIndexingPressure indexingPressure = new InstrumentedIndexingPressure(
-            // Set the coordinating bytes limit high enough to handle all the requests
-            Settings.builder().put(MAX_COORDINATING_BYTES.getKey(), "100kb").build()
-        );
-        final ShardBulkInferenceActionFilter filter = createFilter(threadPool, Map.of(), indexingPressure, useLegacyFormat, inferenceStats);
-
-        final AtomicReference<Exception> expectedDeduplicatedCause = new AtomicReference<>(null);
-        final AtomicInteger deduplicationCount = new AtomicInteger(0);
-        final Consumer<BulkItemRequest> assertBulkItemRequest = (item) -> {
-            BulkItemResponse response = item.getPrimaryResponse();
-            assertNotNull(response);
-            assertTrue(response.isFailed());
-
-            BulkItemResponse.Failure failure = response.getFailure();
-            assertNotNull(failure);
-            assertThat(failure.getStatus(), is(RestStatus.NOT_FOUND));
-            assertThat(failure.getCause(), instanceOf(ResourceNotFoundException.class));
-            assertThat(
-                failure.getCause().getMessage(),
-                containsString("Inference id [missing_inference_id] not found for field [inference_field]")
-            );
-
-            if (expectedDeduplicatedCause.compareAndSet(null, failure.getCause()) == false) {
-                Exception deduplicatedCause = expectedDeduplicatedCause.get();
-                assertThat(deduplicatedCause, sameInstance(failure.getCause()));
-                deduplicationCount.incrementAndGet();
-            }
-        };
-
-        CountDownLatch chainExecuted = new CountDownLatch(1);
-        ActionFilterChain<BulkShardRequest, BulkShardResponse> actionFilterChain = (task, action, request, listener) -> {
-            try {
-                assertNull(request.getInferenceFieldMap());
-                assertThat(request.items().length, equalTo(100));
-
-                for (BulkItemRequest item : request.items()) {
-                    assertBulkItemRequest.accept(item);
-                }
-
-                IndexingPressure.Coordinating coordinatingIndexingPressure = indexingPressure.getCoordinating();
-                assertThat(coordinatingIndexingPressure, notNullValue());
-                verify(coordinatingIndexingPressure, times(100)).increment(eq(1), longThat(l -> l > 0));
-                verify(coordinatingIndexingPressure, times(100)).increment(anyInt(), anyLong());
-
-                // Verify that the coordinating indexing pressure is maintained through downstream action filters
-                verify(coordinatingIndexingPressure, never()).close();
-
-                // Call the listener once the request is successfully processed, like is done in the production code path
-                listener.onResponse(null);
-            } finally {
-                chainExecuted.countDown();
-            }
-        };
-        ActionListener<BulkShardResponse> actionListener = (ActionListener<BulkShardResponse>) mock(ActionListener.class);
-        Task task = mock(Task.class);
-
-        Map<String, InferenceFieldMetadata> inferenceFieldMap = Map.of(
-            "inference_field",
-            new InferenceFieldMetadata("inference_field", "missing_inference_id", new String[] { "inference_field" }, null)
-        );
-
-        BulkItemRequest[] items = new BulkItemRequest[100];
-        for (int i = 0; i < 100; i++) {
-            items[i] = new BulkItemRequest(
-                0,
-                new IndexRequest("index").id(Integer.toString(i)).source("inference_field", randomAlphaOfLengthBetween(3, 20))
-            );
-        }
-
-        BulkShardRequest request = new BulkShardRequest(new ShardId("test", "test", 0), WriteRequest.RefreshPolicy.NONE, items);
-        request.setInferenceFieldMap(inferenceFieldMap);
-        filter.apply(task, TransportShardBulkAction.ACTION_NAME, request, actionListener, actionFilterChain);
-        awaitLatch(chainExecuted, 10, TimeUnit.SECONDS);
-        assertThat(expectedDeduplicatedCause.get(), notNullValue());
-        assertThat(deduplicationCount.get(), equalTo(99));
-
-        IndexingPressure.Coordinating coordinatingIndexingPressure = indexingPressure.getCoordinating();
-        assertThat(coordinatingIndexingPressure, notNullValue());
-        verify(coordinatingIndexingPressure).close();
-    }
-
-    public void testEqualFailureSignatures() {
-        for (int i = 0; i < 10; i++) {
-            final int causeCount = randomIntBetween(0, 5);
-
-            Exception aCause = null;
-            Exception bCause = null;
-            for (int j = 0; j < causeCount; j++) {
-                String message = randomAlphaOfLengthBetween(5, 10);
-                aCause = new RuntimeException(message, aCause);
-                bCause = new RuntimeException(message, bCause);
-            }
-
-            String message = randomAlphaOfLengthBetween(5, 10);
-            ShardBulkInferenceActionFilter.FailureSignature aSignature = new ShardBulkInferenceActionFilter.FailureSignature(
-                new RuntimeException(message, aCause)
-            );
-            ShardBulkInferenceActionFilter.FailureSignature bSignature = new ShardBulkInferenceActionFilter.FailureSignature(
-                new RuntimeException(message, bCause)
-            );
-
-            assertThat(aSignature, equalTo(bSignature));
-        }
-    }
-
-    public void testUnequalFailureSignatures() {
-        final BiFunction<String, Exception, Exception> randomDifference = (m, c) -> switch (randomIntBetween(0, 2)) {
-            case 0 -> new IOException(m, c);
-            case 1 -> {
-                String message = randomValueOtherThan(m, () -> randomAlphaOfLengthBetween(5, 10));
-                yield new RuntimeException(message, c);
-            }
-            case 2 -> {
-                Exception cause = c == null ? new RuntimeException(randomAlphaOfLengthBetween(5, 10)) : null;
-                yield new RuntimeException(m, cause);
-            }
-            default -> throw new IllegalStateException("Unhandled value");
-        };
-
-        for (int i = 0; i < 10; i++) {
-            final int causeCount = randomIntBetween(0, 5);
-            final int inequalityLevel = randomIntBetween(0, causeCount);
-
-            Exception aCause = null;
-            Exception bCause = null;
-            for (int j = causeCount; j > 0; j--) {
-                String message = randomAlphaOfLengthBetween(5, 10);
-                aCause = new RuntimeException(message, aCause);
-                bCause = inequalityLevel == j ? randomDifference.apply(message, bCause) : new RuntimeException(message, bCause);
-            }
-
-            String message = randomAlphaOfLengthBetween(5, 10);
-            ShardBulkInferenceActionFilter.FailureSignature aSignature = new ShardBulkInferenceActionFilter.FailureSignature(
-                new RuntimeException(message, aCause)
-            );
-            ShardBulkInferenceActionFilter.FailureSignature bSignature = new ShardBulkInferenceActionFilter.FailureSignature(
-                inequalityLevel == 0 ? randomDifference.apply(message, bCause) : new RuntimeException(message, bCause)
-            );
-
-            assertThat(aSignature, not(equalTo(bSignature)));
-        }
     }
 
     private static ShardBulkInferenceActionFilter createFilter(

